src/tui/terminal.rs

```diff
@@ -32,20 +35,32 @@ impl DerefMut for TerminalGuard {
 
 impl Drop for TerminalGuard {
     fn drop(&mut self) {
-        let result = exit().wrap_err("Failed to exit terminal during drop");
-        // Ignore the `Result` if we're already panicking; aborting is undesirable
-        if !std::thread::panicking() {
-            result.unwrap();
+        if let Err(error) = exit().wrap_err("Failed to exit terminal during drop") {
+            if std::thread::panicking() {
+                // Ignore the `Result` if we're already panicking; aborting is undesirable.
+                tracing::error!("{error}");
+            } else {
+                panic!("{error}");
+            }
         }
     }
 }
 
+/// Are we currently inside a raw-mode terminal?
+///
+/// This helps us avoid entering or exiting raw-mode twice.
 static INSIDE: AtomicBool = AtomicBool::new(false);
 
+/// Enter raw-mode for the terminal on stdout, set up a panic hook, etc.
+#[instrument(level = "debug")]
 pub fn enter() -> miette::Result<TerminalGuard> {
     use event::KeyboardEnhancementFlags as KEF;
 
-    INSIDE.store(true, Ordering::SeqCst);
+    if INSIDE.load(Ordering::SeqCst) {
+        return Err(miette!(
+            "Cannot enter raw mode; the terminal is already set up"
+        ));
+    }
```

Comment:
Instead of erroring, you could `return Ok(TerminalGuard { .. })` so it's idempotent instead.